freshers-bootcamp/day2/excercise3/solution3.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+)
+
+var balance  uint64 = 500
+
+func withdraw(amount uint64,wg *sync.WaitGroup) {
+	defer wg.Done()
+	var mutex = &sync.Mutex{}
```

Comment:
you have defined mutex inside this function. So mutex will be different in each go routine hence all go routines will be able to take lock and update the balance at the same time.

Reply:
Declared the mutex globally.